### ISSUE TITLE
Enable direct write access to App Catalog through Kubernetes API of the Control Plane

### ISSUE THREAD

**Issue by cokiengchiara** [giantswarm/roadmap#28]
The thought behind this, would be that as App Catalog is mainly green-field, we could enable write access to it earlier than the rest of Giant Swarm.

Structural validation can be done using [OpenAPI schema](https://github.com/giantswarm/giantswarm/issues/5407)
This would require some work on defaulting and validating the (user-facing) CRDs. We already have a [PoC](https://github.com/giantswarm/OPA-PoC/) for that using [OPA](https://github.com/open-policy-agent/opa)
This might also include SSO for the Kubernetes API of Control Planes

## Blocked by

1. Defaulting and Validation is blocked by https://github.com/giantswarm/roadmap/issues/26
2. We have to work with the new WG to define the scope

---

Note: Moved from Managed Services Roadmap 2019 https://github.com/giantswarm/giantswarm/blob/1f0dbd05bd806df2670ccd2a7cc906c1a8990d4f/areas/managed-services/roadmap/2019-ROADMAP-MANAGED-SERVICES.md#stories-for-2019q4 This issue is intentionally left incomplete, Chiara needs to clarify the user story.

**Comment by cokiengchiara** [comment]
Since Vol created the user roles and enabled access control plane API for customers, customer can already create App Catalogs and Apps through k8s api fo control plane. So this is already done?

**Comment by rossf7** [comment]
@cokiengchiara I think this is done. Can we close?

**Comment by cokiengchiara** [comment]
> @cokiengchiara I think this is done. Can we close?

ok!
